### PR TITLE
v3.0 powered by new JS SDKs and C3.js

### PR DESCRIPTION
**Pull request by dustinlarimer**
## What does this PR do? How does it affect users?

This PR replaces [keen-js](https://github.com/keen/keen-js) with [keen-analysis.js@1.1.0](https://github.com/keen/keen-analysis.js) and [keen-dataviz.js@1.0.0](https://github.com/keen/keen-dataviz.js/pull/39), which brings C3.js in as the default underlying visualization library.

screen shot 2016-04-05 at 10 07 54 am

screen shot 2016-04-04 at 3 41 33 pm

screen shot 2016-04-04 at 6 53 13 pm

screen shot 2016-04-04 at 3 16 30 pm

## How should this be tested?

Step through the code line by line. Things to keep in mind as you review:
 - Are there any edge cases not covered by this code?
 - Does this code follow conventions (naming, formatting, modularization, etc) where applicable?

Fetch the branch and/or deploy to staging to test the following:

- [x] Does the code compile without warnings (check shell, console)?
- [x] Do all tests pass?
- [x] Does the UI, pixel by pixel, look exactly as expected (check various screen sizes, including mobile)?
- [x] If the feature makes requests from the browser, inspect them in the Web Inspector. Do they look as expected (parameters, headers, etc)?
- [ ] If the feature sends data to Keen, is the data visible in the project if you run an extraction (include link to collection/query)?
- [ ] If the feature saves data to a database, can you confirm the data is indeed created in the database?

## Related tickets?

### CODE REVIEW

File: client/js/app/app.js [hunk 1]
```diff
@@ -20,15 +23,17 @@ function App(config) {
   this.appDispatcher = AppDispatcher;
   this.targetNode = document.getElementById(config.targetId);
   this.persistence = null;
-  this.client = config.client;
+  this.client = new KeenAnalysis(config.client);
```
Comment (aroc) [inline client/js/app/app.js]: So is the idea to no longer pass in an instantiated client object and instead hard binding to the version of `keen-analysis` we want intentional? I think this makes sense. I don't foresee many scenarios where a user would want to explicitly use a specific version of `keen.js` etc and this reduces risk of breakages with the wrong client version.

File: client/js/app/utils/ProjectUtils.js [hunk 2]
```diff
@@ -126,10 +126,9 @@ module.exports = {
   },
 
   eventsUrl: function(client) {
-    var endpoint = client.config.protocol + "://" + client.config.host;
-    var projectId = client.config.projectId;
-    var masterKey = client.config.masterKey;
-    return endpoint+'/projects/'+projectId+'/events?api_key='+masterKey;
+    return client.url('events', {
```
Comment (aroc) [inline client/js/app/utils/ProjectUtils.js]: Much cleaner.

**Comment by aroc** [comment]
We're going to want to update the `demo/example_index.js` to reflect the changes to how the client is initialized in the `App` constructor. `README` as well.

**Comment by aroc** [comment]
I'm assuming I need to include the `keen-dataviz` `css` file to have this properly work? Just the one from `node_modules/keen-dataviz/dist` should work eh?

**Comment by dustinlarimer** [comment]
@aroc Latest commits include that CSS file and update `demo/example_index.html` :+1:

**Comment by aroc** [comment]
Taking another look now.

**Comment by aroc** [comment]
Ok just tested this out again. The `node_modules` folder isn't accessible by the connect server, so when you load the page `keen-dataviz.css` doesn't get loaded. Screenshot:

image

**Comment by dustinlarimer** [comment]
Hmm.. That shouldn't be happened on the demo page.. Try reinstalling
node_modules quick.

**Comment by aroc** [comment]
Fresh npm install did the trick. This looks good to ship to me. 🚢

**Comment by aroc** [comment]
Can't seem to run tests here, missing CSS important even after fresh `npm install`.

image

**Comment by aroc** [comment]
Nevermind, got them running again after another `npm install`. Weird. Also just back merged with master. :)

**Comment by aroc** [comment]
Ahhhh it's still breaking. It can run tests on their own, but `npm run production` gets that error I posted above.

File: client/js/app/utils/ExplorerUtils.js [hunk 3]
```diff
@@ -166,18 +166,18 @@ module.exports = {
    * @return {undefined}
    */
   runQuery: function(config) {
-    config.client.run(
-      new Keen.Query(config.query.analysis_type, _.omit(config.query, 'analysis_type')),
-      function(err, response) {
+    console.log(config);
+    config.client
+      .query(config.query.analysis_type, _.omit(config.query, 'analysis_type'))
```
Comment (dustinlarimer) [inline client/js/app/utils/ExplorerUtils.js]: @aroc update here causes an issue with some test wiring.

File: test/unit/actions/ExplorerActionsSpec.js [hunk 4]
```diff
@@ -28,103 +29,103 @@ describe('actions/ExplorerActions', function() {
...
-  describe('exec', function () {
-    before(function () {
-      this.client = { run: sinon.spy() };
-      this.getStub = sinon.stub(ExplorerStore, 'get');
-    });
-
-    after(function () {
-      ExplorerStore.get.restore();
-    });
+  // describe('exec', function () {
```
Comment (dustinlarimer) [inline test/unit/actions/ExplorerActionsSpec.js]: @aroc need to fix this up, but not sure how to approach it.

File: client/js/app/utils/ExplorerUtils.js (hunk 3 above)
Comment (dustinlarimer) [inline client/js/app/utils/ExplorerUtils.js]: Ah crap, left a `console.log` in there :(

**Comment by aroc** [comment]
👍
